### ISSUE TITLE
BUG: Pytest does `exit 0` after error during test collection (on PPC)

### ISSUE THREAD

**Issue by h-vetinari** [pytest-dev/pytest#8986]
While packaging scipy on PPC for conda-forge, I got caught out by the test suite failing at correction time, but erroneously reporting `exit 0`: https://github.com/conda-forge/scipy-feedstock/issues/186

```
+ python -c 'import scipy; scipy.test(verbose=2, label='\''full'\'', extra_argv=['\''--durations=50'\''])'
============================= test session starts ==============================
platform linux -- Python 3.9.6, pytest-6.2.4, py-1.10.0, pluggy-0.13.1 -- $PREFIX/bin/python
cachedir: .pytest_cache
rootdir: $SRC_DIR
plugins: xdist-2.3.0, forked-1.3.0
collecting ... collected 46106 items / 1 error / 46105 selected

==================================== ERRORS ====================================
[snip]
=========================== short test summary info ============================
ERROR linalg/tests/test_sketches.py - ValueError: row index exceeds matrix di...
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
========================= 1 error in 713.78s (0:11:53) =========================
+ exit 0
```

Here's the [signature](https://github.com/scipy/scipy/blob/v1.7.1/scipy/_lib/_testutils.py#L27) of `scipy.test` and how this maps into `pytest.main()`.

This should be reproducible - probably only on PPC? - as follows (since the CI passed, the artefacts got published; not yet sure if we'll recall them):
```
conda create -n test_env -c conda-forge python scipy
conda activate test_env
python -c "import scipy; scipy.test(verbose=2, label='full', extra_argv=['--durations=50'])"
```

**Comment by RonnyPfannschmidt** [comment]
The pytest error code is caught in your code and transformed into a boolean, reraise it if you want to put it out

**Comment by h-vetinari** [comment]
OK, fair enough (btw it's not my code, I just help packaging it 🙃). Still it's surprising to me that the `exc.code` is apparently 0? Would there be an argument against setting that code to something non-zero?

**Comment by RonnyPfannschmidt** [comment]
Based on the code the value is ignored, it should be the error value and the tester should actually exit, I'll take a closer look once om back to the computer

**Comment by h-vetinari** [comment]
> Based on the code the value is ignored, it should be the error value and the tester should actually exit, I'll take a closer look once I'm back to the computer

The way I read it, the error code of the exception is checked to equal 0, which seems to happen in this case. Is it intended that some exceptions come with an error code of 0?

**Comment by RonnyPfannschmidt** [comment]
@h-vetinari the code in scipy consumes the pytest error code and returns a boolean, so i consider it a scipy bug

**Comment by h-vetinari** [comment]
> the code in scipy consumes the pytest error code and returns a boolean, so i consider it a scipy bug

Yeah, I can read the code, but the error code is not consumed (as in discarded without effect), "just" turned into a boolean. We can argue (also with the scipy devs) about the reasons for this, but my question remains: is it by design that an exception from pytest has an error code of 0? Would there be an argument against ensuring that it is non-zero?

**Comment by RonnyPfannschmidt** [comment]
uppon reviewing the code - it may be necessary to review where the systemexit is from, i'd like to enable pytest to return the error code there, as well (as far asi i can tell from the main function, it shouldnt error

**Comment by RonnyPfannschmidt** [comment]
ok, pytests own cmdline mains dont raise systemexit, pytest shoud trigger a warning on bad plugins that raise systemexit in cmdline_main
